vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/ssllabs/SSLLabsClient.java

```diff
@@ -0,0 +1,184 @@
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ssllabs;
+
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.hc.core5.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+import org.vividus.util.Sleeper;
+import org.vividus.util.json.JsonUtils;
+
+public class SSLLabsClient
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(SSLLabsClient.class);
+    private static final String SSL_SCAN_FAILURE =
+            "SSL scan has not been performed successfully during specified waiting period";
+    private static final String UNEXPECTED_STATUS_CODE_RECEIVED = "Unexpected status code received: {}";
+    private static final String UNEXPECTED_RESPONSE = "Response: {}";
+    private static final String ERROR_MESSAGE = "Status message '{}' received for host {}";
+    private static final String API_VERSION = "/api/v3";
+    private static final String ANALYZE_CALL = "/analyze?host=%s&fromCache=on&maxAge=1";
+    private static final int TOO_MANY_REQUESTS = 429;
+    private static final int SERVICE_IS_OVERLOADED = 529;
+    private final IHttpClient httpClient;
+    private final JsonUtils jsonUtils;
+    private final String sslLabHost;
+    private final long totalWaitingPeriod;
+    private final long attemptWaitingPeriod;
+
+    public SSLLabsClient(IHttpClient httpClient, JsonUtils jsonUtils, String sslLabHost, long totalWaitingPeriod,
+            long attemptWaitingPeriod)
+    {
+        this.httpClient = httpClient;
+        this.jsonUtils = jsonUtils;
+        this.sslLabHost = sslLabHost;
+        this.totalWaitingPeriod = totalWaitingPeriod;
+        this.attemptWaitingPeriod = attemptWaitingPeriod;
+    }
+
+    public Optional<Grade> performSSLScan(String host)
+    {
+        Optional<AnalyzeResponse> analyzeResponse = waitReadyStatus(host);
+        if (analyzeResponse.isEmpty())
+        {
+            return Optional.empty();
+        }
+        return getMinimumGrade(analyzeResponse.get().getEndpoints());
+    }
+
+    private Optional<AnalyzeResponse> waitReadyStatus(String host)
+    {
+        Optional<AnalyzeResponse> analyzeResponse;
+        long endTime = System.currentTimeMillis() + totalWaitingPeriod;
```

Comment:
reuse `org.vividus.util.wait.DurationBasedWaiter`

Reply:
> reuse `org.vividus.util.wait.DurationBasedWaiter`

I'm not sure how well this class suits us because it always waits `iterationPollingTimeout` regardless of the result of the first attempt to get value

https://github.com/vividus-framework/vividus/blob/d11ab45767d06b4757f2a590003323a457ccb256/vividus-util/src/main/java/org/vividus/util/wait/DurationBasedWaiter.java#L57

Reply:
you can improve `DurationBasedWaiter` to stop immediately if the condition is reached

Reply:
it's better to use a separate PR for this improvement